2021-05-02T14:25:08.3645786+02:00  [INF] Application started. Press Ctrl+C to shut down. (dcaefe54)
2021-05-02T14:25:08.3728156+02:00  [INF] Hosting environment: "Development" (c3307c92)
2021-05-02T14:25:08.3749218+02:00  [INF] Content root path: "C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp" (b5d60022)
2021-05-02T14:31:37.3411470+02:00  [INF] Application started. Press Ctrl+C to shut down. (dcaefe54)
2021-05-02T14:31:37.3517142+02:00  [INF] Hosting environment: "Development" (c3307c92)
2021-05-02T14:31:37.3538578+02:00  [INF] Content root path: "C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp" (b5d60022)
2021-05-02T14:31:47.7767159+02:00 8000035a-0000-ff00-b63f-84710c7967bb [INF] "Andre" logged in with "Google" provider. (107abac5)
2021-05-02T14:32:17.4997369+02:00 800002ec-0003-ff00-b63f-84710c7967bb [INF] User logged in. (fdce0db4)
2021-05-02T14:32:34.4192600+02:00 80000173-0002-fb00-b63f-84710c7967bb [INF] "Andre" logged in with "Google" provider. (107abac5)
2021-05-02T14:32:41.3437265+02:00 8000017b-0002-fb00-b63f-84710c7967bb [INF] Teacher scibby98@gmail.com accessed file of submission 2ebaffa2-be86-44a6-a6e8-f3766dccbbd3 on 02/05/2021 14:32:41 (e4df56c1)
2021-05-02T14:42:37.9266100+02:00  [INF] Application started. Press Ctrl+C to shut down. (dcaefe54)
2021-05-02T14:42:37.9369591+02:00  [INF] Hosting environment: "Development" (c3307c92)
2021-05-02T14:42:37.9392699+02:00  [INF] Content root path: "C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp" (b5d60022)
2021-05-02T14:42:48.0619157+02:00 8000022e-0001-fb00-b63f-84710c7967bb [INF] Teacher scibby98@gmail.com accessed file of submission 2ebaffa2-be86-44a6-a6e8-f3766dccbbd3 on 02/05/2021 14:42:48 (f1acd418)
2021-05-02T14:43:28.5872550+02:00 80000185-0002-fb00-b63f-84710c7967bb [INF] "Andre" logged in with "Google" provider. (107abac5)
2021-05-02T14:44:02.0720094+02:00 8000015d-0000-fe00-b63f-84710c7967bb [INF] User logged in. (fdce0db4)
2021-05-02T14:44:17.1024029+02:00 8000015f-0000-fe00-b63f-84710c7967bb [ERR] An unhandled exception has occurred while executing the request. (48a46595)
Internal.Cryptography.CryptoThrowHelper+WindowsCryptographicException: The parameter is incorrect.
   at Internal.NativeCrypto.CapiHelper.DecryptKey(SafeKeyHandle safeKeyHandle, Byte[] encryptedData, Int32 encryptedDataLength, Boolean fOAEP, Byte[]& decryptedData)
   at System.Security.Cryptography.RSACryptoServiceProvider.Decrypt(Byte[] rgb, Boolean fOAEP)
   at System.Security.Cryptography.RSACryptoServiceProvider.Decrypt(Byte[] data, RSAEncryptionPadding padding)
   at PresentationAssignmentApp.Helpers.CryptographicHelper.AsymmetricDecrypt(Byte[] data, String privateKey) in C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp\Helpers\CryptographicHelper.cs:line 255
   at PresentationAssignmentApp.Controllers.AssignmentsController.ViewFile(String id) in C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp\Controllers\AssignmentsController.cs:line 314
   at lambda_method(Closure , Object , Object[] )
   at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2021-05-02T14:49:38.0536120+02:00  [INF] Application started. Press Ctrl+C to shut down. (dcaefe54)
2021-05-02T14:49:38.0631273+02:00  [INF] Hosting environment: "Development" (c3307c92)
2021-05-02T14:49:38.0654366+02:00  [INF] Content root path: "C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp" (b5d60022)
2021-05-02T14:55:18.5631940+02:00  [INF] Application started. Press Ctrl+C to shut down. (dcaefe54)
2021-05-02T14:55:18.5737626+02:00  [INF] Hosting environment: "Development" (c3307c92)
2021-05-02T14:55:18.5789067+02:00  [INF] Content root path: "C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp" (b5d60022)
2021-05-02T14:55:37.7851042+02:00 80000299-0000-fd00-b63f-84710c7967bb [INF] Bad request when user tried to access a file (0536f105)
2021-05-02T14:55:45.3543907+02:00 800002f5-0003-ff00-b63f-84710c7967bb [INF] Bad request when user tried to access a file (0536f105)
2021-05-02T14:56:43.0915457+02:00 8000029b-0000-fd00-b63f-84710c7967bb [INF] Bad request when user tried to access a file (0536f105)
2021-05-02T14:56:43.2285022+02:00 8000018e-0002-fb00-b63f-84710c7967bb [ERR] An unhandled exception has occurred while executing the request. (48a46595)
Microsoft.AspNetCore.Connections.ConnectionResetException: The client has disconnected
 ---> System.Runtime.InteropServices.COMException (0x800704CD): An operation was attempted on a nonexistent network connection. (0x800704CD)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.IIS.Core.IO.AsyncIOOperation.GetResult(Int16 token)
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContext.ReadBody()
   at System.IO.Pipelines.PipeCompletion.ThrowLatchedException()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
   at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContext.ReadAsync(Memory`1 memory, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.IIS.Core.HttpRequestStream.ReadAsyncInternal(Memory`1 buffer, CancellationToken cancellationToken)
   at System.IO.Pipelines.StreamPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.WebUtilities.FormPipeReader.ReadFormAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Http.Features.FormFeature.InnerReadFormAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenStore.GetRequestTokensAsync(HttpContext httpContext)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.ValidateRequestAsync(HttpContext httpContext)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.Filters.ValidateAntiforgeryTokenAuthorizationFilter.OnAuthorizationAsync(AuthorizationFilterContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2021-05-02T14:56:51.3277500+02:00 800002a1-0000-fd00-b63f-84710c7967bb [INF] "Andre" logged in with "Google" provider. (107abac5)
2021-05-02T14:57:09.5363527+02:00 80000198-0002-fb00-b63f-84710c7967bb [INF] Teacher scibby98@gmail.com accessed file of submission 2ebaffa2-be86-44a6-a6e8-f3766dccbbd3 on 02/05/2021 14:57:09 (34dbb66a)
2021-05-02T14:57:09.5363628+02:00 8000023e-0001-fb00-b63f-84710c7967bb [INF] Teacher scibby98@gmail.com accessed file of submission 2ebaffa2-be86-44a6-a6e8-f3766dccbbd3 on 02/05/2021 14:57:09 (34dbb66a)
2021-05-02T14:57:41.7594145+02:00 800002a9-0000-fd00-b63f-84710c7967bb [INF] User logged in. (fdce0db4)
2021-05-02T14:57:53.1619222+02:00 8000019a-0002-fb00-b63f-84710c7967bb [INF] oasismtofficial@gmail.com tried to access submission with id 2ebaffa2-be86-44a6-a6e8-f3766dccbbd3. Access was denied (ef51d5eb)
2021-05-02T14:58:07.0816343+02:00 8000019e-0002-fb00-b63f-84710c7967bb [ERR] An unhandled exception has occurred while executing the request. (48a46595)
Microsoft.AspNetCore.Connections.ConnectionResetException: The client has disconnected
 ---> System.Runtime.InteropServices.COMException (0x800704CD): An operation was attempted on a nonexistent network connection. (0x800704CD)
   --- End of inner exception stack trace ---
   at Microsoft.AspNetCore.Server.IIS.Core.IO.AsyncIOOperation.GetResult(Int16 token)
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContext.ReadBody()
   at System.IO.Pipelines.PipeCompletion.ThrowLatchedException()
   at System.IO.Pipelines.Pipe.GetReadResult(ReadResult& result)
   at System.IO.Pipelines.Pipe.ReadAsync(CancellationToken token)
   at System.IO.Pipelines.Pipe.DefaultPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContext.ReadAsync(Memory`1 memory, CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Server.IIS.Core.HttpRequestStream.ReadAsyncInternal(Memory`1 buffer, CancellationToken cancellationToken)
   at System.IO.Pipelines.StreamPipeReader.ReadAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.WebUtilities.FormPipeReader.ReadFormAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Http.Features.FormFeature.InnerReadFormAsync(CancellationToken cancellationToken)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgeryTokenStore.GetRequestTokensAsync(HttpContext httpContext)
   at Microsoft.AspNetCore.Antiforgery.DefaultAntiforgery.ValidateRequestAsync(HttpContext httpContext)
   at Microsoft.AspNetCore.Mvc.ViewFeatures.Filters.ValidateAntiforgeryTokenAuthorizationFilter.OnAuthorizationAsync(AuthorizationFilterContext context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
2021-05-02T14:58:24.8370745+02:00 800002ff-0003-ff00-b63f-84710c7967bb [INF] User logged in. (fdce0db4)
2021-05-02T14:58:42.4967401+02:00 80000242-0001-fb00-b63f-84710c7967bb [ERR] An unhandled exception has occurred while executing the request. (48a46595)
Internal.Cryptography.CryptoThrowHelper+WindowsCryptographicException: The parameter is incorrect.
   at Internal.NativeCrypto.CapiHelper.DecryptKey(SafeKeyHandle safeKeyHandle, Byte[] encryptedData, Int32 encryptedDataLength, Boolean fOAEP, Byte[]& decryptedData)
   at System.Security.Cryptography.RSACryptoServiceProvider.Decrypt(Byte[] rgb, Boolean fOAEP)
   at System.Security.Cryptography.RSACryptoServiceProvider.Decrypt(Byte[] data, RSAEncryptionPadding padding)
   at PresentationAssignmentApp.Helpers.CryptographicHelper.AsymmetricDecrypt(Byte[] data, String privateKey) in C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp\Helpers\CryptographicHelper.cs:line 255
   at PresentationAssignmentApp.Controllers.AssignmentsController.ViewFile(String id) in C:\Users\AndrePC\source\repos\SecuringApplicationsAssignment\AssignmentApp\PresentationAssignmentApp\Controllers\AssignmentsController.cs:line 315
   at lambda_method(Closure , Object , Object[] )
   at Microsoft.Extensions.Internal.ObjectMethodExecutor.Execute(Object target, Object[] parameters)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|24_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
